pandas/tests/indexes/test_category.py

```diff
@@ -365,29 +365,36 @@ def test_astype(self):
         tm.assert_index_equal(result, expected)
 
     def test_reindex_base(self):
-
-        # determined by cat ordering
-        idx = self.create_index()
+        # Determined by cat ordering.
+        idx = CategoricalIndex(list("cab"), categories=list("cab"))
         expected = np.arange(len(idx), dtype=np.intp)
 
         actual = idx.get_indexer(idx)
         tm.assert_numpy_array_equal(expected, actual)
 
-        with tm.assert_raises_regex(ValueError, 'Invalid fill method'):
-            idx.get_indexer(idx, method='invalid')
+        with tm.assert_raises_regex(ValueError, "Invalid fill method"):
+            idx.get_indexer(idx, method="invalid")
 
     def test_reindexing(self):
+        np.random.seed(123456789)
```

Comment:
I would leave this, the fact it was random before actually catched a bug

Reply:
You could, but "random" tests are generally not good design-wise.  The fact that it happened to catch a bug doesn't change that IMO.

Reply:
why don't you add a new test for this instead. I would not use random either, rather you can use the specific case that was actually causing the issue.

Reply:
@gfyoung already added the specific case that was failing below without using random data 
(so the remark was only about the original test case whether to keep it random or not)

Reply:
> "random" tests are generally not good design-wise.

that might be in certain cases, but we have a *ton* of random tests in our test suite, and the majority has no seed set